cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java

```diff
@@ -531,30 +529,37 @@ public Deployment generateDeployment(Map<String, String> annotations, boolean is
         List<Container> initContainers = new ArrayList<>(1);
 
         if (rack != null) {
-            Container initContainer = new ContainerBuilder()
-                    .withName(INIT_NAME)
-                    .withImage(initImage)
-                    .withArgs("/opt/strimzi/bin/kafka_init_run.sh")
-                    .withResources(getInitContainerResourceResourceRequirements())
-                    .withEnv(getInitContainerEnvVars())
-                    .withVolumeMounts(VolumeUtils.createVolumeMount(INIT_VOLUME_NAME, INIT_VOLUME_MOUNT))
-                    .withImagePullPolicy(determineImagePullPolicy(imagePullPolicy, initImage))
-                    .withSecurityContext(templateInitContainerSecurityContext)
-                    .build();
-
+            Container initContainer;
+            if (getInitContainerResourceResourceRequirements() != null) {
+                initContainer = new ContainerBuilder()
+                        .withName(INIT_NAME)
+                        .withImage(initImage)
+                        .withArgs("/opt/strimzi/bin/kafka_init_run.sh")
+                        .withResources(getInitContainerResourceResourceRequirements())
+                        .withEnv(getInitContainerEnvVars())
+                        .withVolumeMounts(VolumeUtils.createVolumeMount(INIT_VOLUME_NAME, INIT_VOLUME_MOUNT))
+                        .withImagePullPolicy(determineImagePullPolicy(imagePullPolicy, initImage))
+                        .withSecurityContext(templateInitContainerSecurityContext)
+                        .build();
+            } else {
+                initContainer = new ContainerBuilder()
+                        .withName(INIT_NAME)
+                        .withImage(initImage)
+                        .withArgs("/opt/strimzi/bin/kafka_init_run.sh")
+                        .withEnv(getInitContainerEnvVars())
+                        .withVolumeMounts(VolumeUtils.createVolumeMount(INIT_VOLUME_NAME, INIT_VOLUME_MOUNT))
+                        .withImagePullPolicy(determineImagePullPolicy(imagePullPolicy, initImage))
+                        .withSecurityContext(templateInitContainerSecurityContext)
+                        .build();
+            }
             initContainers.add(initContainer);
         }
 
         return initContainers;
     }
 
     private ResourceRequirements getInitContainerResourceResourceRequirements() {
-        return new ResourceRequirementsBuilder()
-                .addToRequests("cpu", new Quantity("100m"))
-                .addToRequests("memory", new Quantity("128Mi"))
-                .addToLimits("cpu", new Quantity("1"))
-                .addToLimits("memory", new Quantity("256Mi"))
-                .build();
+        return getResources();
```

Comment:
Is this `getInitContainerResourceResourceRequirements` method used also somewhere else? If not, do we really need it? Why not call `getResources()` directly from `getInitContainers()`? It seems not needed if it is not called from anywhere else.

Reply:
Yeah, that was my initial thought. I will check if it is being used somewhere and if not, I will exclude it.

Reply:
Even if it is used somewhere else, you should think if it makes sense to have a separate method for it or not.

Reply:
It is not being used at any other place